Xamarin.Forms.Maps.iOS/MapRenderer.cs

```diff
@@ -20,145 +20,11 @@ namespace Xamarin.Forms.Maps.iOS
-				return null;
-
-			const string defaultPinId = "defaultPin";
-			mapPin = (MKPinAnnotationView)mapView.DequeueReusableAnnotation(defaultPinId);
-			if (mapPin == null)
-			{
-				mapPin = new MKPinAnnotationView(annotation, defaultPinId);
-				mapPin.CanShowCallout = true;
-			}
-
-			mapPin.Annotation = annotation;
-			AttachGestureToPin(mapPin, annotation);
-
-			return mapPin;
-		}
-#if __MOBILE__
-		void AttachGestureToPin(MKPinAnnotationView mapPin, IMKAnnotation annotation)
-		{
-
-			UIGestureRecognizer[] recognizers = mapPin.GestureRecognizers;
-
-			if (recognizers != null)
-			{
-				foreach (UIGestureRecognizer r in recognizers)
-				{
-					mapPin.RemoveGestureRecognizer(r);
-				}
-			}
-
-			Action<UITapGestureRecognizer> action = g => OnClick(annotation, g);
-			var recognizer = new UITapGestureRecognizer(action) { ShouldReceiveTouch = (gestureRecognizer, touch) =>
-			{
-				_lastTouchedView = touch.View;
-				return true;
-			} };
-			List.Add(action);
-			List.Add(recognizer);
-			mapPin.AddGestureRecognizer(recognizer);
-			}
-#else
-		void AttachGestureToPin(MKPinAnnotationView mapPin, IMKAnnotation annotation)
-		{
-			NSGestureRecognizer[] recognizers = mapPin.GestureRecognizers;
-
-			if (recognizers != null)
-			{
-				foreach (NSGestureRecognizer r in recognizers)
-				{
-					mapPin.RemoveGestureRecognizer(r);
-				}
-			}
-
-			Action<NSClickGestureRecognizer> action = g => OnClick(annotation, g);
-			var recognizer = new NSClickGestureRecognizer(action);
-			List.Add(action);
-			List.Add(recognizer);
-			mapPin.AddGestureRecognizer(recognizer);
-
-		}
-#endif
-#if __MOBILE__
-		void OnClick(object annotationObject, UITapGestureRecognizer recognizer)
-#else
-		void OnClick(object annotationObject, NSClickGestureRecognizer recognizer)
-#endif
-		{
-			// https://bugzilla.xamarin.com/show_bug.cgi?id=26416
-			NSObject annotation = Runtime.GetNSObject(((IMKAnnotation)annotationObject).Handle);
-			if (annotation == null)
-				return;
-
-			// lookup pin
-			Pin targetPin = null;
-			for (var i = 0; i < _map.Pins.Count; i++)
-			{
-				Pin pin = _map.Pins[i];
-				object target = pin.Id;
-				if (target != annotation)
-					continue;
-
-				targetPin = pin;
-				break;
-			}
-
-			// pin not found. Must have been activated outside of forms
-			if (targetPin == null)
-				return;
-
-			// if the tap happened on the annotation view itself, skip because this is what happens when the callout is showing
-			// when the callout is already visible the tap comes in on a different view
-			if (_lastTouchedView is MKAnnotationView)
-				return;
-
-			targetPin.SendTap();
-		}
-
-		protected override void Dispose(bool disposing)
-		{
-			if (_disposed)
-			{
-				return;
-			}
-
-			_disposed = true;
-
-			if (disposing)
-			{
-				_map = null;
-				_lastTouchedView = null;
-			}
-
-			base.Dispose(disposing);
-		}
-	}
-
-	public class MapRenderer : ViewRenderer
+	public class MapRenderer : ViewRenderer<Map, MKMapView>
```

Comment:
We can't change the base class definition, it will break every renderer inheriting from it